RF24Network.cpp

```diff
@@ -696,14 +696,39 @@ bool ESBNetwork<radio_t>::write(RF24NetworkHeader& header, const void* message,
 
 /******************************************************************/
 
+template<>
+bool ESBNetwork<RF24>::write(RF24NetworkHeader& header, const void* message, uint16_t len, uint16_t writeDirect)
+{
+    max_frame_size = MAX_FRAME_SIZE;
+    return main_write(header, message, len, writeDirect);
+}
+
+/******************************************************************/
+#if defined NRF52_RADIO_LIBRARY
+template<>
+bool ESBNetwork<nrf_to_nrf>::write(RF24NetworkHeader& header, const void* message, uint16_t len, uint16_t writeDirect)
+{
+    max_frame_size = (uint8_t)NRF_RADIO->PCNF1;
+    if (radio.enableEncryption == true) {
+        max_frame_size -= CCM_IV_SIZE + CCM_COUNTER_SIZE + CCM_MIC_SIZE;
+    }
+    return main_write(header, message, len, writeDirect);
+}
+#endif
+/******************************************************************/
+
 template<class radio_t>
-bool ESBNetwork<radio_t>::write(RF24NetworkHeader& header, const void* message, uint16_t len, uint16_t writeDirect)
+bool ESBNetwork<radio_t>::main_write(RF24NetworkHeader& header, const void* message, uint16_t len, uint16_t writeDirect)
 {
+    max_frame_payload_size = max_frame_size - sizeof(RF24NetworkHeader);
 
 #if defined(DISABLE_FRAGMENTATION)
-    frame_size = rf24_min(len + sizeof(RF24NetworkHeader), MAX_FRAME_SIZE);
+
+    frame_size = rf24_min(len + sizeof(RF24NetworkHeader), max_frame_size);
```

Comment:
This seems to be the only time where `max_frame_payload_size` differs from `max_frame_size`.